test/unit_tests/test_ops.py

```diff
@@ -60,6 +62,19 @@ def test_parametrized_ops(pl_op, braket_gate, angle):
     _assert_decomposition(pl_op, [angle])
 
 
+@patch("braket.pennylane_plugin.ops.np", new=autograd.numpy)
+@pytest.mark.parametrize("pl_op, braket_gate", testdata_parametrized)
+@pytest.mark.parametrize("angle", [(i + 1) * math.pi / 12 for i in range(12)])
+def test_param_shift(pl_op, braket_gate, angle):
```

Comment:
Actually, this looks like the wrong test. The parameter-shift rule does not necessarily give the derivative of a gate matrix; rather, it gives the derivative of an expectation value of a circuit with respect to a parametrized gate. 

A better test would be to use an actual circuit, e.g. by applying the gate to a random starting state, taking the expectation value of a random measurement, and then calculating the derivative of that using the parameter shift rule.

Reply:
Alternatively, if you're trying to test the parameter-shift rule for a gate U(theta), you should consider the observable M(theta) = U(theta) H U^{\dagger}(theta) for a random Hermitian observable H, and then see if dM / d theta satisfies the formula for parameter shift.

Reply:
Yep, echoing what @licedric mentions above!

These tests usually take the form of a integration/functional test, where a small (non-trivial) QNode is created, and the derivative is computed using the parameter-shift rule this.

We then compare this against one or more of:

- The backprop gradient
- The finite difference gradient
- The known, analytic, gradient.

We typically try and go for the third option, since it is surprisingly easy to construct small circuits where you expect a non-zero gradient, but the gradient is actually trivial!

Reply:
Whoops, forgot that we were differentiation a conjugation. Fixed it in now, although rather than full integration tests, I now have unit tests for U^{\dagger}(\theta) H U(\theta).